ale_linters/javascript/eslint.vim

```diff
@@ -49,9 +49,31 @@ function! ale_linters#javascript#eslint#Handle(buffer, lines)
     return l:output
 endfunction
 
-call ale#linter#Define('javascript', {
+function! ale_linters#javascript#eslint#GetExecutable(buffer)
+    let path = finddir('node_modules', ';')
+
+    return path . '/.bin/eslint'
+endfunction
+
+function! ale_linters#javascript#eslint#GetCommand(buffer)
+    let path = finddir('node_modules', ';')
+    let env = 'env NODE_PATH=' . path . ' '
+    let name = fnamemodify(bufname(a:buffer), ':.')
+
+    return env . ale_linters#javascript#eslint#GetExecutable(a:buffer) . ' -f unix --stdin --stdin-filename ' . name
```

Comment:
All of these variables should be scoped, ie `l:var`. Vint will let you know what to change.